include/odp/api/spec/ipsec.h

```diff
@@ -1210,16 +1256,22 @@ typedef struct odp_ipsec_status_t {
  *
  * The operation does packet transformation according to IPSEC standards (see
  * e.g. RFC 4302 and 4303). Resulting packets are well formed, reconstructed
- * original IP packets, with IPSEC headers removed and valid header field values
- * restored. The amount and content of packet data before the IP header is
- * undefined.
+ * original IP (or TFC dummy) packets, with IPSEC headers removed and valid
+ * header field values restored. The amount and content of packet data before
+ * the IP header is undefined. TFC padding may follow the IP packet payload,
+ * in which case packet length is larger than protocol headers indicate.
+ * TFC dummy packets have both IPv4 and IPv6 flags cleared, although L3 offset
+ * is set also for those.
```

Comment:
I think the offset is already known (and possibly set into packet header) before implementation knows a packet is dummy, since that's part of decrypt operation (== start of cipher/clear text data). This spec allows L3 offset to be left at an implementation specific value, e.g. 0 if HW did drop the dummy content, e.g. 28 if HW does not recognize dummy packets (just another decrypted packet from HW point of view).

When dummy content is left there, application can see how much it is and the content. May be that helps in debugging dummy vs non-dummy packet, etc.

Result of an IPSEC decrypt transformation is always the original packet (== what the sender did send). L3 offset points to the first byte of the original data. If dummy data was dropped, then there's no data.